openquakeplatform/openquakeplatform/vulnerability/views.py

```diff
@@ -842,23 +1012,29 @@ def _stream_frag_cont_data(func_obj):
         params.attrib['mean'] = mean_arr[i]
         params.attrib['stddev'] = stddev_arr[i]
 
+    return frmo
 
 
-    return etree.tostring(
-        root, pretty_print=True, xml_declaration=True, encoding="UTF-8")
+def _func_export_as_nrml(request, gen_info_id):
+    if gen_info_id == -1:
+        func_objs = GeneralInformation.objects.filter(cart__username=request.user.username)
+    else:
+        try:
+            func_objs = [ GeneralInformation.objects.get(pk=gen_info_id) ]
+        except:
+            return HttpResponseNotFound(
+                '<h1>Item with id %d not found</h1>' % gen_info_id)
 
+    if gen_info_id == -1 and not func_objs:
+        return HttpResponseNotFound('<h1>The cart is empty.</h1>')
 
-def func_export_as_nrml(request, gen_info_id):
-    try:
-        func_obj = GeneralInformation.objects.get(pk=gen_info_id)
-    except:
-        return HttpResponseNotFound(
-            '<h1>Item with id %d not found</h1>' % gen_info_id)
+    content_disp = ('attachment; filename="%s%s.xml"' %
+                    (slugify(func_objs[0].name), ("_and-others" if
+                                              gen_info_id == -1 else "")))
```

Comment:
Please handle the case in which you have a single curve in the cart and you want to export the whole chart